dotCMS/src/main/java/com/dotmarketing/factories/WebAssetFactory.java

```diff
@@ -459,6 +460,7 @@ public static WebAsset publishAsset(WebAsset currWebAsset, User user) throws Web
 	 * @throws DotSecurityException
 	 */
 	@SuppressWarnings("unchecked")
+	@CloseDBIfOpened
 	public static WebAsset publishAsset(WebAsset currWebAsset, User user, boolean isNewVersion) throws WebAssetException, DotStateException, DotDataException, DotSecurityException {
```

Comment:
keep in mind that the transaction is doing closeAndCommit but it must be just commit, you might be closing twice

Reply:
Let's not put the annotation in this case, since all the transaction handling is ok. Let's just change the `HibernateUtil.closeAndCommitTransaction();` below with `HibernateUtil.commitTransaction();` and leave the rest as it was